server/models/services/TestPlanService.js

```diff
@@ -0,0 +1,63 @@
+const ModelService = require('./ModelService');
+const {
+    TEST_PLAN_ATTRIBUTES,
+    TEST_PLAN_VERSION_ATTRIBUTES,
+    TEST_PLAN_REPORT_ATTRIBUTES
+} = require('./helpers');
+const { TestPlan } = require('../');
+
+const getTestPlans = async (
+    filter = {},
+    testPlanAttributes = TEST_PLAN_ATTRIBUTES,
+    pagination = {},
+    options = {}
+) => {
+    // search and filtering options
+    let where = { ...filter };
+
+    const data = await ModelService.get(
+        TestPlan,
+        where,
+        testPlanAttributes,
+        [
+            {
+                association: 'testPlanVersions',
+                attributes: TEST_PLAN_VERSION_ATTRIBUTES
+            },
+            {
+                association: 'testPlanReports',
+                attributes: TEST_PLAN_REPORT_ATTRIBUTES
+            }
+        ],
+        pagination,
+        options
+    );
+
+    return data.map(d => {
+        const latestTestPlanVersion = d.dataValues.testPlanVersions.sort(
```

Comment:
I believe `sequelize` provides a way for us to also declare the order with the included `association`s, which may be more optimal to do in this case.

I haven’t found a clear example for this version (`v6`) but this link may be helpful https://sequelize.org/docs/v6/core-concepts/model-querying-basics/#ordering.